#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Add a shopt option compat31 to falls back to the behaviour of bash-3.1
# DP: and earlier versions, in that quoting the string argument to the [[
# DP: command's =~ +operator does not force string matching.

--- doc/bashref.texi~	2008-04-14 22:41:58.000000000 +0000
+++ doc/bashref.texi	2008-04-14 22:57:46.000000000 +0000
@@ -3604,6 +3604,11 @@
 command in the same history entry.  This allows
 easy re-editing of multi-line commands.
 
+@item compat31
+If set, bash falls back to the behaviour of bash-3.1 and earlier versions,
+in that quoting the string argument to the @samp{[[} command's @samp{=~}
+operator does @b{not} force string matching.
+
 @item dotglob
 If set, Bash includes filenames beginning with a `.' in
 the results of filename expansion.
